langkit/libmanage.py

```diff
@@ -697,6 +697,7 @@ def gpr_scenario_vars(self, args, build_mode=None,
 
         result = ['-XBUILD_MODE={}'.format(build_mode),
                   '-XLIBRARY_TYPE={}'.format(library_type),
+                  '-XGPR_BUILD={}'.format(library_type),
```

Comment:
Yes, exactly that. 

Right now we're passing -XLIBRARY_TYPE=static and -XXMLADA_BUILD=static, and, if in your environment you have GPR_BUILD=relocatable, you'll get (legitimately) an error of the form:

==============Error messages for file: <path_to>/libgpr_stable/install/share/gpr/gpr.gpr
     9. library project GPR is
                        |
        >>> shared library project "gpr" cannot import static library project "xmlada_dom"

Reply:
I see, thank you for the clarification. This is counter-intuitive: command-line arguments (`-X…`) are supposed to take precedence over environment variables, but because of the nesting of `external` calls in `gpr.gpr`, we now have the `GPR_BUILD` environment variable taking precedence over the `-XLIBRARY_TYPE=static` command-line argument…
Anyway, your change will not harm and will indeed fix spurious inconsistency with dirty environments, so let’s go!